apps/api/src/reservation/core/entities/inventory-item.entity.ts

```diff
@@ -106,4 +108,19 @@ export class InventoryItem extends AggregateRoot {
     this.data = data;
     this.locationId = locationId;
   }
+
+  deleteReservation(reservationId: number, userId: number) {
+    const reservation = this.reservations.find(
+      (res) => res.id === reservationId,
+    );
+
+    if (!reservation) throw new EntityNotFoundException('Reservation');
+
+    if (reservation.userId !== userId)
+      throw new ReservationDeleteException(
```

Comment:
This exception is too generic in my opinion, what do you think about choosing a more descriptive error name?